clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -1691,10 +1691,7 @@ class ConstraintRefersToContainingTemplateChecker
   using inherited::TransformTemplateTypeParmType;
   QualType TransformTemplateTypeParmType(TypeLocBuilder &TLB,
                                          TemplateTypeParmTypeLoc TL, bool) {
-    assert(TL.getDecl()->getDepth() <= TemplateDepth &&
-           "Nothing should reference a value below the actual template depth, "
-           "depth is likely wrong");
-    if (TL.getDecl()->getDepth() != TemplateDepth)
+    if (TL.getDecl()->getDepth() < TemplateDepth)
```

Comment:
I don't see the justification for removing the assert here. When would we otherwise hit it?

Besides, we have the same pattern in `TransformTemplateTypeParmType()` above. Does it also impact anything?

Reply:
(Oh, I saw the issue, just disregard my first question.)

Reply:
I did not notice that before, and we do have similar issue when it comes to NTTP [godbolt](https://godbolt.org/z/ajvfcf4Po)